sys-apps/pciutils/pciutils-3.6.0.recipe

```diff
@@ -0,0 +1,140 @@
+DESCRIPTION="The PCI Utilities are a collection of programs for inspecting \
+and manipulating configuration of PCI devices, all based on a common \
+portable library libpci which offers access to the PCI configuration space \
+on a variety of operating systems.
+
+The utilities include:
+
+- lspci - displays detailed information about all PCI buses and devices in \
+the system
+- setpci - allows reading from and writing to PCI device configuration \
+registers. For example, you can adjust the latency timers with it."
+HOMEPAGE="https://mj.ucw.cz/sw/pciutils/"
+COPYRIGHT="1997-2018 Martin Mares"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="ftp://atrey.karlin.mff.cuni.cz/pub/linux/pci/pciutils-$portVersion.tar.gz"
+CHECKSUM_SHA256="10e9cdeb8d147051dc6aa7f3aa4ae086ba516ce8f81357d6b2d0da9c5923531b"
+PATCHES="pciutils-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+# libpci.so can only be built (for this package) if gcc >= 4.
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+	LIBPCI_SHARED=yes
+else
+	LIBPCI_SHARED=no
+fi
+
+PROVIDES="
+	pciutils$secondaryArchSuffix = $portVersion
+	cmd:lspci$secondaryArchSuffix = $portVersion
+	cmd:setpci$secondaryArchSuffix = $portVersion
+	"
+if [ "$LIBPCI_SHARED" = yes ]; then
+	PROVIDES="$PROVIDES
+		lib:libpci$secondaryArchSuffix = $portVersion compat >= 3
+		"
+fi
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	pciutils_data == $portVersion
+	"
+
+if [ -z "$secondaryArchSuffix" ]; then
+	SUMMARY_data="Utilities for inspecting and manipulating PCI devices \
+(data)"
+	ARCHITECTURES_data="any"
+	PROVIDES_data="
+		pciutils_data = $portVersion
+		"
+	REQUIRES_data=""
+fi
+
+PROVIDES_devel="
+	pciutils${secondaryArchSuffix}_devel = $portVersion
+	devel:libpci$secondaryArchSuffix = $portVersion compat >= 3
+	"
+REQUIRES_devel="
+	pciutils$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:sed
+	"
+
+
+BUILD()
+{
+	make \
+		ZLIB=no \
+		SHARED=$LIBPCI_SHARED \
+		PREFIX=$prefix \
+		SHAREDIR=$dataDir \
+		IDSDIR=$dataDir/pciutils \
+		SBINDIR=$binDir \
+		MANDIR=$manDir \
+		INCDIR=$includeDir \
+		LIBDIR=$libDir
+}
+
+INSTALL()
+{
+	make \
+		ZLIB=no \
+		SHARED=$LIBPCI_SHARED \
+		PREFIX=$prefix \
+		SHAREDIR=$dataDir \
+		IDSDIR=$dataDir/pciutils \
+		SBINDIR=$binDir \
+		MANDIR=$manDir \
+		INCDIR=$includeDir \
+		LIBDIR=$libDir \
+		install install-lib
+
+	# update-pciids is useless because pci.ids is read-only.
+	rm -f $binDir/update-pciids
+	rm -f $manDir/man*/update-pciids.*
+
+	prepareInstalledDevelLib libpci
+	fixPkgconfig
+
+	if [ -z "$secondaryArchSuffix" ]; then
+		maybe_manDir_man7=$manDir/man7
+		packageEntries data \
+			$dataDir
+		sed -i -e "s,$dataDir/,/system/data/,g" \
+			$manDir/man8/lspci.8 \
+			$manDir/man8/setpci.8
+		local include_pci=/system/$relativeOldIncludeDir/pci
```

Comment:
Please use the normal `includeDir`. It may look bad, but it will reflect the correct header location in case the user install the package in user prefix.

Reply:
@alaviss, here `$include_pci` is only used to patch `$manDir/man8/lspci.8` and `$manDir/man8/setpci.8`.
Imo it's OK to have the man pages document the friendly paths.
Shall we keep what we have right now in the recipe about this `include_pci` variable?

Reply:
Nice paths are only good for end-user, for devs it's a nightmare when you can't find what you want where you were told. So I don't agree with the nice paths.

Reply:
Well, I'd say that:
1) even non developers might read the man pages of `lspci` and `setpci`.
2) developers who might install the pciutils package in `/boot/home/config/packages/` (instead of `/boot/system/packages/`) are already aware that anything they would get in `/boot/system/develop/headers/` with "system-wide installs" is to be expected in `/boot/home/config/develop/headers/` with "user-only installs".
3) The real paths are really ugly and do not bring much added value.
4) If I'm not mistaken, the paths shown in the man pages would anyway be wrong if the pciutils package is installed in `/boot/home/config/packages/` instead of `/boot/system/packages/`.

Reply:
Guess I'm fine with whatever paths you put there. But why `relativeOldIncludeDir` though?

Reply:
Good question :)
I don't really know the answer...

I guess that when I added the `$relativeOldIncludeDir` in the recipe I was not yet familiar with all the variables haikuporter puts in `work-x86-*/wrapper-script`, so when I looked for a variable defined as `'develop/headers/x86'`, the first one I saw was `relativeOldIncludeDir`.
BTW, I had found it while viewing the file with vi, but, as we can see with a `grep -n`, the `relativeOldIncludeDir` is much closer to the begining of the file than the `relativeIncludeDir`:
```bash
grep -n "'develop/headers/x86'" wrapper-script 
15:relativeOldIncludeDir='develop/headers/x86'
68:relativeIncludeDir='develop/headers/x86'
```
So I guess I had not even found the shorter variable.

So you are perfectly right. I could have used `relativeIncludeDir` instead.

And yes, we can replace the `relativeOldIncludeDir` by a `relativeIncludeDir` :)